### ISSUE TITLE
[Glyph] Glyph of Rejuvenation

### ISSUE THREAD

**Issue by liverthenoob** [dalaranwow/dalaran-wow#5708]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

Glyph of Rejuvenation (40913) - While your Rejuvenation targets are below 50% health, you will heal them for an additional 50% health.

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Currently this glyph procs a heal equal to 50% of the tick, if the rejuvenation ticks and the target is less than 50% health AFTER the tick.

WORKING Example: Rejuvenation ticks on a target at at 1,000/25,000 hp and ticks for 4,500. The target's hp is at 5,500/20,000 (27%) and therefore the glyph does proc a heal equal to 50% of the tick (2,250) bringing the targets hp to 7,750/20,000 (38%).

ISSUE Example: Rejuvenation ticks on a target at at 9,000/20,000 hp and ticks for 4,500. The target's hp is at 13,500/20,000 (67%) and therefore the glyph does not proc a heal equal to 50% of the tick (2,250) leaving the targets hp at 13,500/20,000 (67%)

**Expected behaviour**: Tell us what should happen instead
The glyph should proc if the tick went off below 50%, regardless of the hp after the tick

ISSUE Example: Rejuvenation ticks on a target at at 9.000/20.000 (45%) for 4,500. The glyph should proc a heal equal to 50% of the tick (2,250) bringing the targets hp to 15,750/20,000 (78%), but it doesnt and the targets hp remains at 13,500/20,000 (67%)

**Steps to reproduce the problem**:

1.  equip glyph of rejuv
2.  get some spellpower gear
3.  get health to ~45%
4. use rejuv and let it tick once then remove it.
5. if it ticked and brought your hp above 50%, observe that the glyph did not proc.

**Include proofs for this behaviour**
https://www.wowhead.com/item=40913/charred-glyph#comments:id=901450
"Now it also procs even when the normal rejuv tick brings your target over 50% (previously it didn't, got fixed)."

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by liverthenoob** [comment]
the issue, as i stated, is in the CheckProc function, which is incorrectly checking the proc without taking into account the targets health BEFORE the heal. this was fixed in live to take into account what the targets health was before the heal, like the wowhead comments say.

I've wrote this code which fixes the issue as described. please feel free to implement it into the server after reviewing it. the only changes i made to trinity core are in the CheckProc function


// 54754 - Glyph of Rejuvenation
class spell_dru_glyph_of_rejuvenation : public AuraScript
{
    PrepareAuraScript(spell_dru_glyph_of_rejuvenation);

    bool Validate(SpellInfo const* /*spellInfo*/) override
    {
        return ValidateSpellInfo({ SPELL_DRUID_GLYPH_OF_REJUVENATION_HEAL });
    }

    bool CheckProc(ProcEventInfo& eventInfo)
    {
	HealInfo* healInfo = eventInfo.GetHealInfo();
	if (!healInfo || !healInfo->GetHeal())
		return;

	Unit* target = eventInfo.GetProcTarget();
	int32 const halfHp = (target->GetMaxHealth() / 2.f);
	int32 const currentHealth = target->GetHealth();

	if(currentHealth - healInfo->GetHeal() <= halfHp){
		return True;
	} else{
		return False;
	}
    }

    void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
    {
        PreventDefaultAction();
        HealInfo* healInfo = eventInfo.GetHealInfo();
        if (!healInfo || !healInfo->GetHeal())
            return;

        CastSpellExtraArgs args(aurEff);
        args.AddSpellBP0(CalculatePct(healInfo->GetHeal(), aurEff->GetAmount()));
        eventInfo.GetActor()->CastSpell(eventInfo.GetProcTarget(), SPELL_DRUID_GLYPH_OF_REJUVENATION_HEAL, args);
    }

    void Register() override
    {
        DoCheckProc += AuraCheckProcFn(spell_dru_glyph_of_rejuvenation::CheckProc);
        OnEffectProc += AuraEffectProcFn(spell_dru_glyph_of_rejuvenation::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
    }
};

**Comment by Lothloryen** [comment]
Hello there,

Thank you for your report and proposal, really appreciated. This code doesn't work for us but, it gave me an idea about what the problem is. It has been fixed, so it will be working after the next scheduled maintenance.
